Add IONOS API support

lbrocke opened the pull request:
This PR adds support for the [IONOS DNS API](https://developer.hosting.ionos.de/docs). ACME v2 wildcard certificates are supported, [all tests are passing on my fork](https://github.com/lbrocke/acme.sh/actions/runs/499459718).

Please let me know if any changes are desired. Otherwise I'm happy to update the docs and create an issue for bug reports.

Closes #3284

Neilpang commented on dnsapi/dns_ionos.sh:
```diff
@@ -0,0 +1,184 @@
+
+_get_root() {
+  domain=$1
+  i=2
```
`i`  should start from `1`

lbrocke replied:
Fixed

Neilpang commented:
>> Otherwise I'm happy to update the docs and create an issue for bug reports.

Please go ahead.

lbrocke commented:
> > > Otherwise I'm happy to update the docs and create an issue for bug reports.
> 
> Please go ahead.

I updated the documentation (see [here](https://github.com/acmesh-official/acme.sh/wiki/dnsapi#129-using-the-ionos-domain-api)) and created an issue for bug reports (#3379). The desired change is also now implemented.
